### ISSUE TITLE
Training and Inference with Custom Dataset

### ISSUE THREAD

**Issue by SelimSavas** [chensnathan/YOLOF#43]
Hi, I want to use custom dataset (in COCO format) but I have some questions. 

1) What do I do after I open a folder named "datasets" and put my dataset in it? When I start the training it doesn't see my dataset and gives an error. It automatically sees coco2017 and some other famous datasets, but how can I train on my own dataset? Can you tell me the scripts I need to add?
2) What should the hierarchy of the dataset be like? Should the annotation files be in a separate folder or inside the image folders?
3) I also have test set (unseen data, in addition to training and validation sets). How do I edit the config file to inference on this test dataset?
4) How can I use COCO Evaluator with my custom dataset? 

I know the following commands but I don't understand the usage with custom dataset. 
(1) python ./tools/train_net.py --num-gpus 8 --config-file ./configs/yolof_R_50_C5_1x.yaml 
(2) python ./tools/train_net.py --num-gpus 8 --config-file ./configs/yolof_R_50_C5_1x.yaml --eval-only MODEL.WEIGHTS /path/to/checkpoint_file

**Comment by SelimSavas** [comment]
@chensnathan
